source/grid/grid_out.cc

```diff
@@ -2526,6 +2529,107 @@ void GridOut::write_vtu (const Triangulation<dim,spacedim> &tria,
 
 
 
+template <int dim, int spacedim>
+void GridOut::write_debug_vtu_per_processor (const Triangulation<dim,spacedim> &tria,
+                                             std::string              filename,
+                                             bool                     view_levels,
+                                             bool                     include_artificial) const
+{
+  std::vector<DataOutBase::Patch<dim,spacedim> > patches;
+  unsigned int n_datasets=4;
+  std::vector<std::string> data_names;
+  data_names.push_back("subdomain");
+  data_names.push_back("lvlsubdomain");
+  data_names.push_back("level");
+  data_names.push_back("proc_writing");
+
+  unsigned int n_q_points=GeometryInfo<dim>::vertices_per_cell;
+
+  typename Triangulation<dim, spacedim>::cell_iterator cell, endc;
+  for (cell=tria.begin(), endc=tria.end();
+       cell != endc; ++cell)
+    {
+      if (!include_artificial && cell->level_subdomain_id() ==
+          numbers::artificial_subdomain_id)
+        continue;
+      if (!view_levels && cell->has_children())
+        continue;
+
+      DataOutBase::Patch<dim,spacedim> patch;
+      patch.data.reinit(n_datasets, n_q_points);
+      patch.points_are_available = false;
+
+      for (unsigned int vertex=0; vertex<n_q_points; ++vertex)
+        {
+          patch.vertices[vertex] = cell->vertex(vertex);
+          patch.data(0,vertex) = cell->level();
+          if (!cell->has_children())
+            patch.data(1,vertex) =
+              (double)static_cast<int>(cell->subdomain_id());
+          else
+            patch.data(1,vertex) = -1.0;
+          patch.data(2,vertex) =
+            (double)static_cast<int>(cell->level_subdomain_id());
+          patch.data(3,vertex) =
+            (double)static_cast<int>(Utilities::MPI::this_mpi_process(MPI_COMM_WORLD));
+        }
+
+      for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+        patch.neighbors[f] = numbers::invalid_unsigned_int;
+      patches.push_back(patch);
+    }
+
+  const std::string new_file = (filename + ".proc" +
+                                Utilities::int_to_string (tria.locally_owned_subdomain(), 4) +
+                                ".vtu");
+  std::ofstream out(new_file.c_str());
+  std::vector<std_cxx1x::tuple<unsigned int, unsigned int, std::string> > vector_data_ranges;
+  DataOutBase::VtkFlags flags;
+  DataOutBase::write_vtu (patches,
+                          data_names,
+                          vector_data_ranges,
+                          flags,
+                          out);
+  //creates .pvtu record
+  if (Utilities::MPI::this_mpi_process(MPI_COMM_WORLD) == 0)
+    {
+      std::vector<std::string> filenames;
+      unsigned int indx = filename.size();
+
+      //.pvtu file must exclude path/to/filename
```

Comment:
that comment is a bit weird, shouldn't it be something like ".pvtu needs to reference the files without a relative path because it will be written in the same directory. For this, remove any paths from filename."